src/main/java/com/igot/cb/community/service/impl/CommunityManagementServiceImpl.java

```diff
@@ -284,4 +290,152 @@ public ApiResponse update(JsonNode communityDetails, String authToken) {
         }
     }
 
+    @Override
+    public ApiResponse joinAndUnjoinCommunity(Map<String, Object> request, String authToken) {
+        log.info("CommunityEngagementService:joinAndUnjoinCommunity:joining");
+        ApiResponse response = ProjectUtil.createDefaultResponse(Constants.API_COMMUNITY_UPDATE);
+        String error = validateJoinPayload(request);
+        if (StringUtils.isNotBlank(error)) {
+            return returnErrorMsg(error, HttpStatus.BAD_REQUEST, response);
+        }
+
+        try {
+            String userId = accessTokenValidator.verifyUserToken(authToken);
+            if (StringUtils.isBlank(userId) || userId.equalsIgnoreCase(
+                Constants.UNAUTHORIZED_USER)) {
+                response.getParams().setErrMsg(Constants.USER_ID_DOESNT_EXIST);
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                return response;
+            }
+            String communityId = (String) request.get(Constants.COMMUNITY_ID);
+            Optional<CommunityEntity> optComment = communityEngagementRepository.findByCommunityIdAndIsActive(
+                communityId, true);
+            if (optComment == null || !optComment.isPresent() || optComment.get().getData()
+                .isEmpty()) {
+                response.setResponseCode(HttpStatus.BAD_REQUEST);
+                response.getParams().setErr(error);
+                return response;
+            }
+            Map<String, Object> propertyMap = new HashMap<>();
+            propertyMap.put(Constants.USER_ID, userId);
+            propertyMap.put(Constants.CommunityId, communityId);
+            List<Map<String, Object>> userCommunityDetails = cassandraOperation.getRecordsByPropertiesWithoutFiltering(
+                Constants.KEYSPACE_SUNBIRD, Constants.USER_COMMUNITY_TABLE, propertyMap, null, 1);
+            if (CollectionUtils.isEmpty(userCommunityDetails)) {
```

Comment:
this implementation is not correct... based on incoming operation you need to either join the community or leave from community... 
but here if the record exist - it is removing the user and if record not exist it adds the user... 

what will happen if user clicks twice or the api called twice by mistake... ?